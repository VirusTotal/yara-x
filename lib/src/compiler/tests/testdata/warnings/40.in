rule test_1 {
    condition:
        uint8(0) == 0x1FF
}

rule test_2 {
    condition:
        int8(0) == -129
}

rule test_3 {
    condition:
        uint16(0) == 0x1FFFF
}

rule test_4 {
    condition:
        int16(0) == -32769
}

rule test_5 {
    condition:
        uint32(0) == 0x1FFFFFFFF
}

rule test_6 {
    condition:
        int32(0) == -2147483649
}

rule test_7 {
    condition:
        uint8be(0) == 0x1FF
}

rule test_8 {
    condition:
        int8be(0) == -129
}

rule test_9 {
    condition:
        uint16be(0) == 0x1FFFF
}

rule test_10 {
    condition:
        int16be(0) == -32769
}

rule test_11 {
    condition:
        uint32be(0) == 0x1FFFFFFFF
}

rule test_12 {
    condition:
        int32be(0) == -2147483649
}

