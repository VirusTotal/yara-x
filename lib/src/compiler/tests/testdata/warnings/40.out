warning[unsatisfiable_expr]: unsatisfiable expression
 --> line:5:9
  |
5 |         uint8(0) == 0x1FF
  |         --------    ----- this integer is outside the range [0,255]
  |         |
  |         this expression is an integer in the range [0,255]
  |
warning[unsatisfiable_expr]: unsatisfiable expression
  --> line:10:9
   |
10 |         int8(0) == -129
   |         -------    ---- this integer is outside the range [-128,127]
   |         |
   |         this expression is an integer in the range [-128,127]
   |
warning[unsatisfiable_expr]: unsatisfiable expression
  --> line:15:9
   |
15 |         uint16(0) == 0x1FFFF
   |         ---------    ------- this integer is outside the range [0,65535]
   |         |
   |         this expression is an integer in the range [0,65535]
   |
warning[unsatisfiable_expr]: unsatisfiable expression
  --> line:20:9
   |
20 |         int16(0) == -32769
   |         --------    ------ this integer is outside the range [-32768,32767]
   |         |
   |         this expression is an integer in the range [-32768,32767]
   |
warning[unsatisfiable_expr]: unsatisfiable expression
  --> line:25:9
   |
25 |         uint32(0) == 0x1FFFFFFFF
   |         ---------    ----------- this integer is outside the range [0,4294967295]
   |         |
   |         this expression is an integer in the range [0,4294967295]
   |
warning[unsatisfiable_expr]: unsatisfiable expression
  --> line:30:9
   |
30 |         int32(0) == -2147483649
   |         --------    ----------- this integer is outside the range [-2147483648,2147483647]
   |         |
   |         this expression is an integer in the range [-2147483648,2147483647]
   |
warning[unsatisfiable_expr]: unsatisfiable expression
  --> line:35:9
   |
35 |         uint8be(0) == 0x1FF
   |         ----------    ----- this integer is outside the range [0,255]
   |         |
   |         this expression is an integer in the range [0,255]
   |
warning[unsatisfiable_expr]: unsatisfiable expression
  --> line:40:9
   |
40 |         int8be(0) == -129
   |         ---------    ---- this integer is outside the range [-128,127]
   |         |
   |         this expression is an integer in the range [-128,127]
   |
warning[unsatisfiable_expr]: unsatisfiable expression
  --> line:45:9
   |
45 |         uint16be(0) == 0x1FFFF
   |         -----------    ------- this integer is outside the range [0,65535]
   |         |
   |         this expression is an integer in the range [0,65535]
   |
warning[unsatisfiable_expr]: unsatisfiable expression
  --> line:50:9
   |
50 |         int16be(0) == -32769
   |         ----------    ------ this integer is outside the range [-32768,32767]
   |         |
   |         this expression is an integer in the range [-32768,32767]
   |
warning[unsatisfiable_expr]: unsatisfiable expression
  --> line:55:9
   |
55 |         uint32be(0) == 0x1FFFFFFFF
   |         -----------    ----------- this integer is outside the range [0,4294967295]
   |         |
   |         this expression is an integer in the range [0,4294967295]
   |
warning[unsatisfiable_expr]: unsatisfiable expression
  --> line:60:9
   |
60 |         int32be(0) == -2147483649
   |         ----------    ----------- this integer is outside the range [-2147483648,2147483647]
   |         |
   |         this expression is an integer in the range [-2147483648,2147483647]
   |
